lib/logflare/endpoints/cache.ex

```diff
@@ -170,29 +177,32 @@ defmodule Logflare.Endpoints.Cache do
     {:noreply, %{state | query_tasks: tasks}}
   end
 
+  def handle_info(:shutdown, state) do
+    {:stop, :normal, state}
+  end
+
   def handle_info({from_task, {:ok, results}}, state) do
     tasks = Enum.reject(state.query_tasks, &(&1.pid == from_task))
-    {:noreply, %{state | cached_result: results, query_tasks: tasks}}
+
+    :timer.cancel(state.shutdown_timer)
+    timer = refresh(proactive_querying_ms(state))
+
+    {:noreply, %{state | cached_result: results, query_tasks: tasks, shutdown_timer: timer}}
```

Comment:
This means that if results are received, the Cache will never shutdown through the `:shutdown` callback and will continuously keep refreshing...

Reply:
also rather confusing to set the `timer` to `:shutdown_timer`, doesn't quite make sense.
 setting `shutdown_timer: nil` and having separate key like `:refresh_timer` would be clearer